### PR TITLE
Complete are_all_terminal_heating_loops_attached_to_boiler

### PR DESCRIPTION
**Pull request by weilixu**

### CODE REVIEW

File: rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_attached_to_boiler.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    -------
+    bool
+        True: fluid loop associated with the heating_from_loop associated with each terminal unit is attached to a boiler
+        False: otherwise
+    """
+    are_all_terminal_heating_loops_attached_to_boiler_flag = True
+    boilers = find_all("$.boilers[*]", rmi_b)
+    loop_boiler_dict = dict()
+    for boiler_b in boilers:
+        loop_id = getattr_(boiler_b, "boiler", "loop")
```
Comment (yunjoonjung-PNNL) [inline rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_attached_to_boiler.py:28]: Could you check if ```boiler``` is meant to be ```boilers```?

**Review by yunjoonjung-PNNL — APPROVED** [review]
Approved with one minor comment for double-checking purposes.

File: rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_attached_to_boiler.py (hunk 1 above)
Comment (weilixu) [reply rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_attached_to_boiler.py:28]: Which `boiler` are you referring to?

Comment (charliepnnl) [reply rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_attached_to_boiler.py:28]: @yunjoonjung-PNNL I suspect you are referring to the "boiler" as the second argument on line 28.  That argument is intended to be a string representing the type of entity the first argument represents. That string appears in the exception that is raised if there is no loop in boiler_b.

File: rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_attached_to_boiler.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    for terminal_b_id in terminal_unit_id_list:
+        terminal_b = find_exactly_one_with_field_value(
+            "$.buildings[*].building_segments[*].zones[*].terminals",
+            "id",
+            terminal_b_id,
+            rmi_b,
+        )
+        heating_from_loop = terminal_b.get("heating_from_loop")
+        if heating_from_loop:
```
Comment (charliepnnl) [inline rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_attached_to_boiler.py:41]: Could we call this variable, _heating_from_loop_id_? Or maybe just _fluid_loop_id_?

File: rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_attached_to_boiler.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                heating_from_loop,
+                rmi_b,
+            )
+            if (
+                getattr_(fluid_loop, "fluid loop", "type") != FLUID_LOOP_TYPE.HEATING
+                or heating_from_loop not in loop_boiler_dict.keys()
+            ):
+                are_all_terminal_heating_loops_attached_to_boiler_flag = False
+                break
+        else:
```
Comment (charliepnnl) [inline rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_attached_to_boiler.py:54]: The RDS does not specify what is to be done in this case, I think.

**Review by charliepnnl — APPROVED** [review]
Approved with comments.

File: rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_attached_to_boiler.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        List of terminal units IDs
+
+    Returns
+    -------
+    bool
+        True: fluid loop associated with the heating_from_loop associated with each terminal unit is attached to a boiler
+        False: otherwise
+    """
+    are_all_terminal_heating_loops_attached_to_boiler_flag = True
+    loop_boiler_id_list = [
```
Comment (charliepnnl) [inline rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_attached_to_boiler.py:30]: The RDS should be updated to match.

**Review by charliepnnl — APPROVED** [review]
I think the RDS should be updated to reflect the use of `loop_boiler_id_list` instead of `loop_boiler_dict`.

**Pull request merged**